src/sentry/incidents/logic.py

```diff
@@ -553,6 +562,19 @@ def create_alert_rule(
 
     resolution = get_alert_resolution(time_window, organization)
 
+    if (sensitivity or seasonality) and not (sensitivity and seasonality):
+        raise ValidationError("Must choose both sensitivity and seasonality")
+
+    if detection_type == AlertRuleDetectionType.STATIC:
+        if comparison_delta is not None or sensitivity is not None or seasonality is not None:
```

Comment:
we're checking if any of them are set... 🤔 
this seems unnecessary.

can we not just have a validation here for dynamic, then `else` if any of them are set raise the error?

maybe something like
```
if detection_type == AlertRuleDetectionType.DYNAMIC:
    if not (sensitivity or seasonality):
        raise ...
else:
    if (sensitivity or seasonality):
        raise ...

if detection_type == AlertRuleDetectionType.PERCENT:
    if comparison_delta is None:
        raise ...
else:
    if comparison_delta is not None:
        raise ...
```

or something like this?

Reply:
This is a lot prettier 😅